nixos/modules/programs/hyprland.nix

```diff
@@ -0,0 +1,87 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+with lib; let
+  cfg = config.programs.hyprland;
+
+  defaultHyprlandPackage = pkgs.hyprland.override {
+    enableXWayland = cfg.xwayland.enable;
+    hidpiXWayland = cfg.xwayland.hidpi;
+    nvidiaPatches = cfg.nvidiaPatches;
+  };
+in
+{
+  options.programs.hyprland = {
+    enable = mkEnableOption null // {
+      description = mdDoc ''
+        Hyprland, the dynamic tiling Wayland compositor that doesn't sacrifice on its looks.
+
+        You can manually launch Hyprland by executing "exec Hyprland" on a TTY.
+
+        A configuration file will be generated in ~/.config/hypr/hyprland.conf.
+        See <https://wiki.hyprland.org> for more information.
+      '';
+    };
+
+    package = mkOption {
+      type = types.path;
+      default = defaultHyprlandPackage;
+      defaultText = literalExpression ''
+        pkgs.hyprland.override {
+          enableXWayland = cfg.xwayland.enable;
+          hidpiXWayland = cfg.xwayland.hidpi;
+          nvidiaPatches = cfg.nvidiaPatches;
```

Comment:
Should this be explicit in documenting `config.programs.hyprland.*` here rather than just `cfg.*`?